.github/workflows/libcxx-build-and-test.yaml

```diff
@@ -201,10 +201,20 @@ jobs:
           os: macos-15
         - config: apple-configuration
           os: macos-15
+        # TODO: These jobs are intended to test back-deployment (building against ToT libc++ but running against an
+        #       older system-provided libc++.dylib). Doing this properly would require building the test suite on a
+        #       recent macOS using a recent Clang (hence recent Xcode), and then running the actual test suite on an
+        #       older mac. We could do that by e.g. sharing artifacts between the two jobs.
+        #
+        #       However, our Lit configuration currently doesn't provide a good way to do that in a batch, so our only
+        #       alternative is to actually build on the same host that we're going to run on. Sadly, that doesn't work
+        #       since older macOSes don't support newer Xcodes. For now, we run the "backdeployment" jobs on recent
+        #       macOS versions as a way to avoid rotting that configuration, but it doesn't provide a lot of additional
+        #       coverage.
         - config: apple-system
-          os: macos-13
+          os: macos-15
         - config: apple-system-hardened
-          os: macos-13
+          os: macos-15
     runs-on: ${{ matrix.os }}
```

Comment:
I think we also need to change line 223 `xcode-version: 'latest'`. IMO this should be a fixed version like '14.3.1' (or what the appropriate version is). Using `latest` has some drawbacks: 
- the release branch may suddenly switch to different compiler when a new beta release is available. In our release branch we don't want to use newer compilers.
- updating the compiler usually breaks builds; typically we need to add some `XFAIL` or `UNSUPPORTED` to our test suite. This should be a concussions decision by the libc++ developers and not depend on an upstream change.

Reply:
I totally agree. I thought about doing that change in a follow-up patch to decouple it from the change in coverage for backdeployment.

Reply:
That's fine by me.